src/Services/Gateways/BaseGateway.php

```diff
@@ -78,7 +78,8 @@ public function isAvailableForCountry($country)
         $siteCurrencyMatchesCountry = $siteCurrency === $localCurrency;
 
         return $countryIsAccepted
-            && ($siteCurrencyIsGlobal || $siteCurrencyMatchesCountry);
+            && ($siteCurrencyIsGlobal || $siteCurrencyMatchesCountry)
+            && self::acceptsCurrency($siteCurrency);
```

Comment:
Create a var $currencyIsAccepted, receive the value of your static function call and then add it after $countryIsAccepted one...